test/fw/ptl/utils/pbs_dshutils.py

```diff
@@ -2070,61 +2070,84 @@ def create_temp_dir(self, hostname=None, suffix='', prefix='PtlPbs',
         :type suffix: str
         :param prefix: the directory name will begin with this prefix
         :type prefix: str
-        :param dir: the directory will be created in this directory
-        :type dir: str or None
-        :param uid: Optional username or uid of temp directory owner
-        :param gid: Optional group name or gid of temp directory
+        :param dirname: the directory will be created in this directory
+        :type dirname: str or None
+        :param asuser: Optional username of temp directory owner
+        :param asgroup: Optional group name of temp directory
                     group owner
         :param mode: Optional mode bits to assign to the temporary
                      directory
         :param level: logging level, defaults to INFOCLI2
         """
         preserve = False
         if mode is not None:
-            preserve = True
-        else:
-            mode = 0o755
+            if asuser is None:
+                preserve = True
+            else:
+                preserve = False
+        current_user_info = self.get_id_info(self.get_current_user())
+        uid = current_user_info['uid']
+        gid = current_user_info['gid']
+        if asuser is not None:
+            uid = self.get_id_info(str(asuser))['uid']
+        if asgroup is not None:
+            gid = grp.getgrnam(str(asgroup))[2]
         # create a temp dir as current user
         tmpdir = tempfile.mkdtemp(suffix, prefix)
         if dirname is not None:
             dirname = str(dirname)
-            self.chmod(path=tmpdir, mode=mode)
+            self.chmod(path=tmpdir, mode=0o755)
             self.run_copy(hostname, src=tmpdir, dest=dirname, runas=asuser,
-                          recursive=True,
+                          recursive=True, gid=gid, uid=uid,
                           preserve_permission=preserve, level=level)
+            if mode is not None:
+                self.chmod(hostname, path=dirname, mode=mode, runas=asuser)
+            else:
+                self.chmod(hostname, path=dirname, mode=0o755, runas=asuser)
+
             tmpdir = dirname + tmpdir[4:]
 
         # if temp dir to be created on remote host
         if not self.is_localhost(hostname):
             if asuser is not None:
-                # by default mkstemp creates dir with 0600 permission
+                # by default mkstemp creates dir with 0700 permission
                 # to create dir as different user first change the dir
                 # permission to 0755 so that other user has read permission
-                self.chmod(path=tmpdir, mode=mode)
+                self.chmod(path=tmpdir, mode=0o755)
```

Comment:
The comment says 0700, but the code says 0o755

Reply:
Right, what happens is that mkdtemp creates the directory (line 2090), and that directory is created as 700 but we need to change that to ensure the directory can be copied as a different user. This is the reason it is 755.
I will update the comment to show octal numbers.